services/BasicTypes.proto

```diff
@@ -99,6 +100,50 @@ message TokenTransferList {
     repeated NftTransfer nftTransfers = 3; // Applicable to tokens of type NON_FUNGIBLE_UNIQUE. Multiple list of NftTransfers, each of which has a sender and receiver account, including the serial number of the NFT
 }
 
+/* Representing the percentage of tokens to be transferred as part of the CustomFee */
+message Fraction {
+    uint64 numerator = 1; // numerator of the fraction
+    uint64 denominator = 2; // denominator of the fraction
+}
+
+/* Fraction of the Token to be collected along with its floor if the transfer amount is low*/
+message FractionalFee {
+    Fraction fractionOfUnitsToCollect = 1; // percentage of the Tokens involved in the transfer to be charged as fee
+    uint64 minimumUnitsToCollect = 2; // minimum token units to be charged as fee
+    google.protobuf.UInt64Value maximumUnitsToCollect = 3; // maximum token units to be charged as fee
```

Comment:
HIP 18 doesn't define a maximum unit for collection yet. If this is enabled, it would result in less token units in royalty percentage. For eg, a 10% royalty on 100 token transfer would lead to 10 tokens as custom fees, whereas transferring an amount of 100000 with a maximum limit of 1000 would lead to only 1% of total transfer as custom fees.